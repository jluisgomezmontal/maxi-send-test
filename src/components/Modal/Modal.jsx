/* eslint-disable react/prop-types */
import { PropTypes } from "prop-types";

export const Modal = ({ dataModal, setModal, modal }) => {
  return (
    <div className="popup shadow-lg p-3 mb-5 bg-white rounded">
      <h1>Title: {dataModal.title}</h1>
      <div className="d-flex gap-5">
        <img src={dataModal.images.PosterArt.url} alt={dataModal.title} />
        <div>
          <p>Release Year: {dataModal.releaseYear}</p>
          <p>Description: {dataModal.description}</p>
          <button onClick={() => setModal(!modal)} className="btn btn-danger">
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

Modal.propTypes = {
  dataModal: PropTypes.object.isRequired,
  setModal: PropTypes.func.isRequired,
  modal: PropTypes.bool.isRequired,
};
