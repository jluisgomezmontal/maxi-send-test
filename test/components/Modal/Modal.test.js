/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import { Modal } from "../../../src/components/Modal/Modal";

describe("testing <Modal/>", () => {
  const modal = false;
  const setModal = jest.fn();
  const dataModal = {
    title: "Wolf Creek",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    programType: "series",
    images: {
      PosterArt: {
        url: "http://localhost/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg",
        width: 1000,
        height: 1500,
      },
    },
    releaseYear: 2016,
  };
  test("should render propertly", () => {
    render(
      <BrowserRouter>
        <Modal dataModal={dataModal} setModal={setModal} modal={modal} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("should title, img and Release Year", () => {
    render(
      <BrowserRouter>
        <Modal dataModal={dataModal} setModal={setModal} modal={modal} />
      </BrowserRouter>
    );

    expect(screen.getByText(`Title: ${dataModal.title}`)).toBeInTheDocument();
    expect(screen.getByRole("img").src).toBe(dataModal.images.PosterArt.url);
    expect(
      screen.getByText(`Release Year: ${dataModal.releaseYear}`)
    ).toBeInTheDocument();
  });
});
