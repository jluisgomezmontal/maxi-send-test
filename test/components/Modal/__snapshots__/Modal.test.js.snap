// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Header/> should render propertly 1`] = `
<div>
  <nav
    class="navbar navbar-dark bg-primary justify-content-between"
  >
    <div
      class="container-xl"
    >
      <a
        class="navbar-brand "
        href="/"
      >
        <h1
          class="fw-bold p-0 m-0 my-1"
        >
          DEMO Streaming
        </h1>
      </a>
      <form
        class="form-inline"
      >
        <a
          class="text-white text-decoration-none"
          href="/"
        >
          Log in
        </a>
        <a
          class="btn_header"
          href="/"
        >
          Start your free trial
        </a>
      </form>
    </div>
  </nav>
  <div
    class="bg-steel text-white"
  >
    <div
      class="container-xl"
    >
      <h2
        class="py-2"
      >
        Popular
         
        Titles
      </h2>
    </div>
  </div>
</div>
`;

exports[`testing <Modal/> should render propertly 1`] = `
<div>
  <div
    class="popup shadow-lg p-3 mb-5 bg-white rounded"
  >
    <h1>
      Title: 
      Wolf Creek
    </h1>
    <div
      class="d-flex gap-5"
    >
      <img
        alt="Wolf Creek"
        src="http://localhost/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
      />
      <div>
        <p>
          Release Year: 
          2016
        </p>
        <p>
          Description: 
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
        <button
          class="btn btn-danger"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
